ext/LinearSolveEnzymeExt.jl

```diff
@@ -9,6 +9,52 @@ using Enzyme
 
 using EnzymeCore
 
+function EnzymeCore.EnzymeRules.forward(func::Const{typeof(LinearSolve.init)}, ::Type{RT}, prob::EnzymeCore.Annotation{LP}, alg::Const; kwargs...) where {RT, LP <: LinearSolve.LinearProblem}
+    @assert !(prob isa Const)
+    res = func.val(prob.val, alg.val; kwargs...)
+    if RT <: Const
+        return res
+    end
+    dres = func.val(prob.dval, alg.val; kwargs...)
+    dres.b .= res.b == dres.b ? zero(dres.b) : dres.b
+    dres.A .= res.A == dres.A ? zero(dres.A) : dres.A
+    if RT <: DuplicatedNoNeed
+        return dres
+    elseif RT <: Duplicated
+        return Duplicated(res, dres)
```

Comment:
This doesn't handle batching atm

Reply:
tried to address this in 585cbb8. We can  make a followup PR for batch support.